[EventHub/ServiceBus] Breaking change Alert for EventHub ServiceBus

schaudhari6254888 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
- Added Breaking Change Warning for parameter datatype change. 
- cmdlets - `New-AzEventHub`
datatype of parameter `IdentityType` will change from `CaptureIdentityType` to `String`
datatype of parameter `CleanupPolicy` will change from `CleanupPolicyRetentionDescription` to `String`

- cmdlets - `New-AzEventHub` `Set-AzEventHub`
datatype of parameter `Encoding` will change from `EncodingCaptureDescription` to `String`

- cmdlets - `Set-AzEventHub` `New-AzEventHub` `Set-AzServiceBusQueue` `Set-AzServiceBusSubscription` `Set-AzServiceBusTopic`
datatype of parameter `Status` will change from `EntityStatus` to `String`.


- cmdlets - `New-AzEventHubSchemaGroup`
datatype of parameter `SchemaCompatibility` will change from `SchemaCompatibility` to `String`
datatype of parameter `SchemaType` will change from `SchemaType` to `String`

- cmdlets -  `New-AzEventHubAuthorizationRule` `Set-AzEventHubAuthorizationRule` `Set-AzServiceBusAuthorizationRule` `New-AzServiceBusAuthorizationRule`
datatype of parameter `Rights` will change from `AccessRights[]` to `String[]`

- cmdlets - `New-AzEventHubIPRuleConfig` `New-AzServiceBusIPRuleConfig`
datatype of parameter `Action` will change from `NetworkRuleIPAction` to `String`

- cmdlets - `New-AzEventHubKey` `New-AzServiceBusKey`
datatype of parameter `KeyType` will change from `KeyType` to `String`

- cmdlets - `New-AzEventHubNamespace` `New-AzServiceBusNamespace` `Set-AzServiceBusNamespace`
datatype of parameter `SkuName` will change from `SkuName` to `String`

- cmdlets - `New-AzEventHubThrottlingPolicyConfig` 
datatype of parameter `MetricId` will change from `MetricId` to `String`

- cmdlets - `Set-AzEventHubNamespace` `New-AzEventHubNamespace` `Set-AzEventHubNetworkRuleSet` `New-AzServiceBusNamespace` `Set-AzServiceBusNamespace` `Set-AzServiceBusNetworkRuleSet`
datatype of parameter `PublicNetworkAccess` will change from `PublicNetworkAccess` to `String`

- cmdlets - `Set-AzEventHubNetworkRuleSet` `Set-AzServiceBusNetworkRuleSet`
datatype of parameter `DefaultAction` will change from `DefaultAction` to `String`

- cmdlets - `New-AzEventHubNamespace` `Set-AzEventHub` `Set-AzEventHubNamespace`
`New-AzServiceBusNamespace` `Set-AzServiceBusNamespace` 
datatype of parameter `IdentityType` will change from `ManagedServiceIdentityType` to `String`



<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

schaudhari6254888 commented:
We are migrating the EventHub & ServiceBus cmdlets using Autorest V4.
So some datatype are changing. This PR send Notification alerts for this datatype changes in upcoming breaking change release.
that Migration V4 PR is also there for reference.
https://github.com/Azure/azure-powershell/pull/24070

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: CaptureIdentityType
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
The current Az.EventHub version is 4.2.0 on PSGallery. So the deprecated EventHub version should be 5.0.0 since you don't have preview. Please update the version number accordingly.

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: CleanupPolicyRetentionDescription
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
Please update the version number.

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: EntityStatus
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
Please update the version number.

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: EncodingCaptureDescription
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
Please update the version number.

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: SchemaCompatibility
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
Please update the version number.

vidai-msft commented on src/EventHub/EventHub.Autorest/README.md:
```diff
@@ -460,3 +457,64 @@ directive:
+      breaking-change:
+        old-parameter-type: SchemaType
+        new-parameter-type: String
+        deprecated-by-version: 7.0.0
```
Please update the version number.

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubAuthorizationRule.ps1:
```diff
@@ -64,6 +64,7 @@ function New-AzEventHubAuthorizationRule{
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.AccessRights[]]
         # The rights associated with the rule.
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("Rights","12.0.0", "7.0.0","2024-05-21" )]
```
Please update the version number.
```suggestion
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("Rights","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubIPRuleConfig.ps1:
```diff
@@ -29,6 +29,7 @@ function New-AzEventHubIPRuleConfig{
 
 		[Parameter(HelpMessage = "The IP Filter Action")]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.NetworkRuleIPAction]
+		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("Action","12.0.0", "7.0.0","2024-05-21" )]
```
```suggestion
		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("Action","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubKey.ps1:
```diff
@@ -65,6 +65,7 @@ function New-AzEventHubKey{
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.KeyType]
         # The access key to regenerate.
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("KeyType","12.0.0", "7.0.0","2024-05-21" )]
```
```suggestion
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("KeyType","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubNamespace.ps1:
```diff
@@ -83,6 +83,7 @@ function New-AzEventHubNamespace{
         [Parameter(HelpMessage = "Type of managed service identity.")]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.ManagedServiceIdentityType]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("IdentityType","12.0.0", "7.0.0","2024-05-21" )]
```
```suggestion
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("IdentityType","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubNamespace.ps1:
```diff
@@ -109,11 +110,13 @@ function New-AzEventHubNamespace{
         [Parameter(HelpMessage = "The minimum TLS version for the cluster to support, e.g. '1.2'")]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.SkuName]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("SkuName","12.0.0", "7.0.0","2024-05-21" )]
```
```suggestion
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("SkuName","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubNamespace.ps1:
```diff
@@ -109,11 +110,13 @@ function New-AzEventHubNamespace{
         [Parameter(HelpMessage = "This determines if traffic is allowed over public network. By default it is enabled.")]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.PublicNetworkAccess]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("PublicNetworkAccess","12.0.0", "7.0.0","2024-05-21" )]
```
```suggestion
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("PublicNetworkAccess","12.0.0", "5.0.0","2024-05-21" )]
```

vidai-msft commented on src/EventHub/EventHub.Autorest/custom/New-AzEventHubThrottlingPolicyConfig.ps1:
```diff
@@ -35,7 +35,8 @@ function New-AzEventHubThrottlingPolicyConfig{
         [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Support.MetricId]
 		# Metric Id on which the throttle limit should be set, MetricId can be discovered by hovering over Metric in the Metrics section of Event Hub Namespace inside Azure Portal.
-        ${MetricId}
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("MetricId","12.0.0", "7.0.0","2024-05-21" )]
```
Please update the version number.

vidai-msft commented:
@schaudhari6254888 All the version numbers are incorrect. Please fix them all. In addition, please also find the following action items.
1. Select the target release for this PR. If you plan to merge into the main branch and release with Az. Then you should choose General release
2. Leave a comment here as change log message.

schaudhari6254888 commented:
> @schaudhari6254888 All the version numbers are incorrect. Please fix them all. In addition, please also find the following action items.
> 
> 1. Select the target release for this PR. If you plan to merge into the main branch and release with Az. Then you should choose General release
> 2. Leave a comment here as change log message.

Hey @vidai-msft , 
1. we want to merge this breaking notification alert in this march (bcs we are catching the upcoming breaking change cycle)
2. I have updated the description about the breaking change.

vidai-msft commented:
@schaudhari6254888 I did not point out all the incorrect deprecated version numbers in the comments because there were too many and made the page too large. Can you please check these versions again and fixed them all?

schaudhari6254888 commented:
> @schaudhari6254888 I did not point out all the incorrect deprecated version numbers in the comments because there were too many and made the page too large. Can you please check these versions again and fixed them all?

So this is the Az.EventHub current Version. So in Deprecated msg I have changed it to 5.0.0
![image](https://github.com/Azure/azure-powershell/assets/111409694/aa1f5d09-4b38-4403-9b03-5c4a3a80db9d)


For Az.Servicebus
![image](https://github.com/Azure/azure-powershell/assets/111409694/add06191-b407-4f67-b473-15d3c45e77dc)
This its the current version. So in deprecated msg for servicebus I have put 4.0.0

vidai-msft commented:
Thank you @schaudhari6254888 . I did not realize you changed two different modules.

vidai-msft reviewed (approved):
LGTM
